src/dom/context2d.mjs

```diff
@@ -1,10 +1,12 @@
-export default function(width, height, dpi) {
-  if (dpi == null) dpi = devicePixelRatio;
-  var canvas = document.createElement("canvas");
-  canvas.width = width * dpi;
-  canvas.height = height * dpi;
+export default function (width, height, options = {}) {
+  const {scale = devicePixelRatio, ...contextOptions} = !isNaN(options)
+    ? {...(options != null && {scale: options})}
+    : options;
+  const canvas = document.createElement("canvas");
+  canvas.width = width * scale;
+  canvas.height = height * scale;
   canvas.style.width = width + "px";
-  var context = canvas.getContext("2d");
-  context.scale(dpi, dpi);
+  const context = canvas.getContext("2d", !options ? options : contextOptions);
```

Comment:
```suggestion
  const context = canvas.getContext("2d", contextOptions);
```

Reply:
This is meant to pass an explicit _null_ to the constructor, as per https://github.com/observablehq/stdlib/pull/273#discussion_r772564108 ; if we drop that requirement then the simpler code works for me.

Reply:
In that case you'll want to do an explicit check for null: `options === null`

Reply:
done